### ISSUE TITLE
Simplifier Bug in `ReadByte`

### ISSUE THREAD

**Issue by d-xo** [ethereum/hevm#304]
Discovered during a ci run:

```
ReadByte (Lit 0xd1615e4bf65612fa4570939e9fc8a9ba7331e8637f8bb1659c231798d39dc212) (WriteByte (And (SHA256 (WriteByte (Lit 0x2d) (LitByte 108) (ConcreteBuf "k\132\250\229\244\NUL%\DC1\a\SUB\141\170\244\192\159z3\246(re\NUL\192\149G\206\DC3\ETX\US\GSv"))) (Lit 0x15)) (LitByte 172) (ConcreteBuf "\166LK1hG\168\STXt\181\139\216#\215\GS\208Okrn\200\151\tp\239\165(T"))
```

does not simplify correctly. should reproduce with `--quickcheck-replay=429143`

**Comment by msooseth** [comment]
A shorter version that also triggers the bug: `ReadByte (Lit 0xd1615e4bf65612fa4570939e9fc8a9ba7331e8637f8bb1659c231798d39dc212) (WriteByte (And (SHA256 (ConcreteBuf "")) (Lit 0x15)) (LitByte 0) (ConcreteBuf ""))
`

**Comment by msooseth** [comment]
Fixed in #309 Yay!
